### PR TITLE
Develop upstream sync 230821

### PR DESCRIPTION
**Pull request by weihanmines**

### CODE REVIEW

**Comment by weihanmines** [comment]
retest Ubuntu-CPU please

**Comment by weihanmines** [comment]
retest Ubuntu-GPU-single please

File: tensorflow/compiler/xla/stream_executor/rocm/rocm_driver.cc
Comment (i-chaochen) [inline tensorflow/compiler/xla/stream_executor/rocm/rocm_driver.cc:1]: we don't need this macro anymore due to https://github.com/openxla/xla/pull/5076

**Comment by i-chaochen** [comment]
You might need to cherry-pick this PR https://github.com/openxla/xla/pull/5146

**Comment by weihanmines** [comment]
> You might need to cherry-pick this PR [openxla/xla#5146](https://github.com/openxla/xla/pull/5146)

Why do you postpone error checking here https://github.com/openxla/xla/pull/5146/files#diff-3f21301ab0bb47f6adc6ea9a97ab53e68ff45a3f7b2af3fe4d6f7038b7419ea3R580?

File: tensorflow/compiler/xla/stream_executor/rocm/rocm_driver.cc
Comment (weihanmines) [reply tensorflow/compiler/xla/stream_executor/rocm/rocm_driver.cc:1]: removed the preprocessor in the latest pr.

**Comment by i-chaochen** [comment]
> > You might need to cherry-pick this PR [openxla/xla#5146](https://github.com/openxla/xla/pull/5146)
> 
> Why do you postpone error checking here https://github.com/openxla/xla/pull/5146/files#diff-3f21301ab0bb47f6adc6ea9a97ab53e68ff45a3f7b2af3fe4d6f7038b7419ea3R580?

@draganmladjenovic

**Comment by draganmladjenovic** [comment]
> > > You might need to cherry-pick this PR [openxla/xla#5146](https://github.com/openxla/xla/pull/5146)
> > 
> > 
> > Why do you postpone error checking here https://github.com/openxla/xla/pull/5146/files#diff-3f21301ab0bb47f6adc6ea9a97ab53e68ff45a3f7b2af3fe4d6f7038b7419ea3R580?
> 
> @draganmladjenovic

In order to get more descriptive error message. Otherwise we fail too early.

**Review by jayfurmanek — APPROVED** [review]
LGTM

**Comment by jayfurmanek** [comment]
@draganmladjenovic  Note the new "clang official" bazelrc config here.

**Comment by draganmladjenovic** [comment]
@jayfurmanek We wish to do the same? So we can use -config=rocm_clang w/o custom bazelrc?

**Pull request merged**